Docs: Add section about derived fields for Loki (#20648)
grafana/grafana: docs/sources/features/datasources/loki.md

| _URL_           | The URL of the Loki instance, e.g., `http://localhost:3100`                                                                                   |
| _Maximum lines_ | Upper limit for number of log lines returned by Loki (default is 1000). Decrease if your browser is sluggish when displaying logs in Explore. |

### Derived fields

> Derived fields configuration allow you to add additional fields parsed from the log message and optionally add link that uses the value of the field. This can be used for example to link to trace in your tracing backend from your logs or user profile page if userId is present int the log line. They are then showed in the [log details](/features/explore/#labels-and-parsed-fields)

Reviewer: ```suggestion
The Derived Fields configuration allows you to: 
* Add fields parsed from the log message. 
* Add a link that uses the value of the field. 

You can use this functionality to link to your tracing backend directly from your logs, or link to a user profile page if a userId is present in the log line. 

They are shown in the [log details](/features/explore/#labels-and-parsed-fields).
```

Reviewer: Unclear antecedent. Is "They" in the last paragraph here the links, user ID, or something else? Please clarify.

Author: It refers to the links themselves so should probably be something like `These links will be shown in ...`

Author: Changed to `These links will be shown in`

{{< docs-imagebox img="/img/docs/v65/loki_derived_fields.png" class="docs-image--no-shadow" caption="Screenshot of the derived fields configuration" >}}
Each derived field consists of:

> - Name: will be shown in the log details as label

Reviewer: ```suggestion
- **Name:** Shown in the log details as a label.
```

> - Regex: regex pattern that will be run on the log message and should capture some part of it to be used as the value of the new field. Can only contain single capture group.

Reviewer: ```suggestion
- **Regex:** A Regex pattern that runs on the log message and captures part of it to as the value of the new field. Can only contain capture a single group.
```

> - URL: a url template that will be used to construct a link next to the field value in log details. Use special `${__value.raw}` to insert the value of the field.

Reviewer: ```suggestion
- URL: A URL template used to construct a link next to the field value in log details. Use special `${__value.raw}` to insert the value of the field.
```

Reviewer: Is the second sentence in this bullet instructing the reader to **use** the command to insert the value or saying that the template **uses** the command to insert the value?

Author: It is used for interpolating the real value into the URL string. Like: `http://trace-service/trace/${__value.raw}` will later replace `${__value.raw}` with the value of the field. So maybe something like:
> Use special `${__value.raw}` value in your template to interpolate the real field value into your URL template.

Reviewer: Good idea!

> You can also use a debug section to see what does your fields extract and how the URL is interpolated. Click on Show example log message button to show text area where you can write or copy paste a log message.

Reviewer: ```suggestion
You can use a debug section to see what your fields extract and how the URL is interpolated. Click **Show example log message** to show the text area where you can enter a log message.
```

Reviewer: Do you mean you can use a debug section in the logs? Unclear where we are putting the debug section.

Author: The debug section is right under the derived fields form. There is a screenshot that shows `Show example log message` button next to the form so I think it may be enough.
